main/tests/server/src/com/google/refine/model/recon/StandardReconConfigTests.java

```diff
@@ -337,9 +337,8 @@ public void reconTest() throws Exception {
         try (MockWebServer server = new MockWebServer()) {
             server.start();
             HttpUrl url = server.url("/openrefine-wikidata/en/api");
-            // FIXME: Retry doesn't currently work, but should be tested
-//            server.enqueue(new MockResponse().setResponseCode(503)); // service overloaded
-            server.enqueue(new MockResponse().setBody(reconResponse));
+            server.enqueue(new MockResponse().setResponseCode(503)); // service initially overloaded
```

Comment:
Do we expect the recon request to fail on the first try, so there should be something like "Retrying recon queries:..." on the console? I ran `StandardReconConfigTests.reconTest()` locally, but there's no such information 

It's worth mentioning that duplicating `server.enqueue(new MockResponse().setResponseCode(503));` for several times here did cause "Retrying recon queries:..." to be printed, not sure why.

Reply:
> It's worth mentioning that duplicating server.enqueue(new MockResponse().setResponseCode(503)); for several times here did cause "Retrying recon queries:..." to be printed, not sure why.

All right, after taking a look at [HttpClient.java](https://github.com/OpenRefine/OpenRefine/blob/master/main/src/com/google/refine/util/HttpClient.java), I found that a request will be retried up to 3 times if the server returns 503, since we do use the [ExponentialBackoffRetryStrategy](https://github.com/OpenRefine/OpenRefine/blob/68b148d7755c829cf42fadba5b821c1ba193d050/main/src/com/google/refine/util/HttpClient.java#L185) when building the http client.

The change to this testcase doesn't cause the newly added retry logic to be executed as I understand and tested, but the logic does work as I tested. Anyway, since this is a minimal change and a temporary fix, it should be ok to merge it into the master branch :)

Reply:
Ah ok then I need to have a second look at it - I think the ExponentialBackoffRetryStrategy is not used for POST requests, so I thought the change to the testcase did test my changes. If that's not the case then I need to revise my PR.